[Feature][Connector-V2]  jdbc connector supports Kingbase database

NickCodeJourney opened the pull request:
close #4512
close #4758
<!--

Thank you for contributing to SeaTunnel! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GITHUB issue](https://github.com/apache/seatunnel/issues).

  - Name the pull request in the form "[Feature] [component] Title of the pull request", where *Feature* can be replaced by `Hotfix`, `Bug`, etc.

  - Minor fixes should be named following this pattern: `[hotfix] [docs] Fix typo in README.md doc`.

-->

## Purpose of this pull request

<!-- Describe the purpose of this pull request. For example: This pull request adds checkstyle plugin.-->

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

Commit:
:sparkles: 支持kingbase

Commit:
:sparkles: 支持kingbase

Commit:
:sparkles: support kingbase

hailin0 commented:
why close？

Commit:
:fire:

NickYoungPeng commented:
> why close？

some code is not push, wait a min

Commit:
:fire:

Commit:
:fire:

Commit:
:art:

Commit:
:art:

Hisoka-X commented on seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/kingbasees/KingbaseESTypeMapper.java:
```diff
@@ -0,0 +1,152 @@
+                return PrimitiveByteArrayType.INSTANCE;
+            case KB_BYTEA_ARRAY:
+                return ArrayType.BYTE_ARRAY_TYPE;
+            case KB_SMALLINT:
```
SMALLINT should map to short

Commit:
:sparkles: KB_SMALLINT map SHORT_TYPE

hailin0 commented:
please add e2e testcase
https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e

Commit:
:art: format code

NickYoungPeng commented:
> please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e


What  os  environment does e2e need？

NickCodeJourney committed:
Merge branch 'apache:dev' into dev

NickCodeJourney committed:
Merge branch 'apache:dev' into dev

NickCodeJourney committed:
Merge branch 'apache:dev' into dev

Commit:
:sparkles: 支持kerberos认证

NickCodeJourney committed:
Merge branch 'apache:dev' into dev

NickYoungPeng commented:
> please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e

can you help me? tell me some skill how can i do it

Commit:
:sparkles: 支持kerberos认证

NickCodeJourney committed:
Merge branch 'apache:dev' into dev

Commit:
:white_check_mark: add e2e kingbasees

Commit:
:bug: remove data type that does not exist

Commit:
:sparkles: init create table  ddl

Commit:
:sparkles: add insert sql

Commit:
:sparkles: add insert sql

Commit:
:sparkles: add insert sql

NickCodeJourney commented:
> please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e

need all success？
<img width="529" alt="Snipaste_2023-07-07_09-57-44" src="https://github.com/apache/seatunnel/assets/72905543/826d67fb-2844-470d-9796-053c51ebbb71">

Commit:
:sparkles: add KMlicense.dat

Commit:
:fire: Delete kudu related

Commit:
:rewind: revert

Commit:
:pencil2:

NickCodeJourney commented:
@hailin0 @EricJoy2048 @Hisoka-X please review

NickCodeJourney commented:
> > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> 
> need all success？ <img alt="Snipaste_2023-07-07_09-57-44" width="529" src="https://user-images.githubusercontent.com/72905543/251616238-826d67fb-2844-470d-9796-053c51ebbb71.png">

hello @TyrantLucifer ,when i test using spark container 
`Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968"
	at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63)
	at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
	at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`

NickCodeJourney commented:
> > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > 
> > 
> > need all success？ <img alt="Snipaste_2023-07-07_09-57-44" width="529" src="https://user-images.githubusercontent.com/72905543/251616238-826d67fb-2844-470d-9796-053c51ebbb71.png">
> 
> hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`

<img width="1720" alt="image" src="https://github.com/apache/seatunnel/assets/72905543/a62cf1b3-4785-4a54-9db1-077fc63d3bb3">
  in local env is ok,I suspect something is wrong with the container

Commit:
:memo: Update the release-note.

Commit:
:art: format code

Commit:
:art: add License head

Commit:
:art: remove KMlicense.dat and add javadoc

CheneyYin commented:
> > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > 
> > > 
> > > need all success？ <img alt="Snipaste_2023-07-07_09-57-44" width="529" src="https://user-images.githubusercontent.com/72905543/251616238-826d67fb-2844-470d-9796-053c51ebbb71.png">
> > 
> > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> 
> <img width="1720" alt="image" src="https://github.com/apache/seatunnel/assets/72905543/a62cf1b3-4785-4a54-9db1-077fc63d3bb3">
>   in local env is ok,I suspect something is wrong with the container

[This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)

NickCodeJourney commented:
> > > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > > 
> > > > 
> > > > need all success？ <img alt="Snipaste_2023-07-07_09-57-44" width="529" src="https://user-images.githubusercontent.com/72905543/251616238-826d67fb-2844-470d-9796-053c51ebbb71.png">
> > > 
> > > 
> > > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> > 
> > 
> > <img alt="image" width="1720" src="https://user-images.githubusercontent.com/72905543/251966367-a62cf1b3-4785-4a54-9db1-077fc63d3bb3.png">
> >   in local env is ok,I suspect something is wrong with the container
> 
> [This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)

I tried the method in the link, but I couldn't resolve the issue.It seems unrelated to the Kingbase database container.I used a local Spark for testing, but the Kingbase database was still the one in the container, and it worked successfully.

CheneyYin commented:
> > > > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > > > 
> > > > > 
> > > > > need all success？ <img alt="Snipaste_2023-07-07_09-57-44" width="529" src="https://user-images.githubusercontent.com/72905543/251616238-826d67fb-2844-470d-9796-053c51ebbb71.png">
> > > > 
> > > > 
> > > > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> > > 
> > > 
> > > <img alt="image" width="1720" src="https://user-images.githubusercontent.com/72905543/251966367-a62cf1b3-4785-4a54-9db1-077fc63d3bb3.png">
> > >   in local env is ok,I suspect something is wrong with the container
> > 
> > [This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)
> 
> I tried the method in the link, but I couldn't resolve the issue.It seems unrelated to the Kingbase database container.I used a local Spark for testing, but the Kingbase database was still the one in the container, and it worked successfully.

I mean, you can try to configure LC=en_US.UTF-8 for spark container.  Because in the container, LC=POSIX usually.

NickYoungPeng commented:
Yes, I have added the spark container, but it can't be solved

> 2023年7月12日 22:50，Chengyu Yan ***@***.***> 写道：
> 
> 
> please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> 
> need all success？  <https://user-images.githubusercontent.com/72905543/251616238-826d67fb-2844-470d-9796-053c51ebbb71.png>
> hello @TyrantLucifer <https://github.com/TyrantLucifer> ,when i test using spark container Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
> 
>  <https://user-images.githubusercontent.com/72905543/251966367-a62cf1b3-4785-4a54-9db1-077fc63d3bb3.png> in local env is ok,I suspect something is wrong with the container
> This link may help you to fix client encoding error <https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/>
> I tried the method in the link, but I couldn't resolve the issue.It seems unrelated to the Kingbase database container.I used a local Spark for testing, but the Kingbase database was still the one in the container, and it worked successfully.
> 
> I mean, you can try to configure LC=en_US.UTF-8 for spark container. Because in the container, LC=POSIX usually.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/apache/seatunnel/pull/4803#issuecomment-1632676368>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ARMHGR4DILYITJPMWBSBSRTXP22SHANCNFSM6AAAAAAYKDSXDA>.
> You are receiving this because you modified the open/close state.
>

NickCodeJourney commented:
> > > > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > > > 
> > > > > 
> > > > > need all success？ <img alt="Snipaste_2023-07-07_09-57-44" width="529" src="https://user-images.githubusercontent.com/72905543/251616238-826d67fb-2844-470d-9796-053c51ebbb71.png">
> > > > 
> > > > 
> > > > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> > > 
> > > 
> > > <img alt="image" width="1720" src="https://user-images.githubusercontent.com/72905543/251966367-a62cf1b3-4785-4a54-9db1-077fc63d3bb3.png">
> > >   in local env is ok,I suspect something is wrong with the container
> > 
> > 
> > [This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)
> 
> I tried the method in the link, but I couldn't resolve the issue.It seems unrelated to the Kingbase database container.I used a local Spark for testing, but the Kingbase database was still the one in the container, and it worked successfully.

Yes, I have added the spark container, but it can't be solved.

NickCodeJourney commented:
> > > > > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > > > > 
> > > > > > 
> > > > > > need all success？ <img alt="Snipaste_2023-07-07_09-57-44" width="529" src="https://user-images.githubusercontent.com/72905543/251616238-826d67fb-2844-470d-9796-053c51ebbb71.png">
> > > > > 
> > > > > 
> > > > > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> > > > 
> > > > 
> > > > <img alt="image" width="1720" src="https://user-images.githubusercontent.com/72905543/251966367-a62cf1b3-4785-4a54-9db1-077fc63d3bb3.png">
> > > >   in local env is ok,I suspect something is wrong with the container
> > > 
> > > 
> > > [This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)
> > 
> > 
> > I tried the method in the link, but I couldn't resolve the issue.It seems unrelated to the Kingbase database container.I used a local Spark for testing, but the Kingbase database was still the one in the container, and it worked successfully.
> 
> Yes, I have added the spark container, but it can't be solved.

<img width="865" alt="image" src="https://github.com/apache/seatunnel/assets/72905543/2c577bdb-0c4e-4e97-af9c-9360d8885f06">

CheneyYin commented:
> > > > > > > > please add e2e testcase https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e
> > > > > > > 
> > > > > > > 
> > > > > > > need all success？ <img alt="Snipaste_2023-07-07_09-57-44" width="529" src="https://user-images.githubusercontent.com/72905543/251616238-826d67fb-2844-470d-9796-053c51ebbb71.png">
> > > > > > 
> > > > > > 
> > > > > > hello @TyrantLucifer ,when i test using spark container `Exception in thread "main" org.apache.seatunnel.core.starter.exception.CommandExecuteException: ErrorCode:[API-01], ErrorDescription:[Configuration item validate failed] - PluginName: jdbc, PluginType: source, Message: com.kingbase8.util.KSQLException: FATAL: invalid value for parameter "client_encoding": "ANSI_X3.4-1968" at org.apache.seatunnel.core.starter.spark.command.SparkTaskExecuteCommand.execute(SparkTaskExecuteCommand.java:63) at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40) at org.apache.seatunnel.core.starter.spark.SeaTunnelSpark.main(SeaTunnelSpark.java:35) at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) at java.lang.reflect.Method.invoke(Method.java:498) at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52) at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:958) at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:180) at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:203) at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:90) at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046) at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055) at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)`
> > > > > 
> > > > > 
> > > > > <img alt="image" width="1720" src="https://user-images.githubusercontent.com/72905543/251966367-a62cf1b3-4785-4a54-9db1-077fc63d3bb3.png">
> > > > >   in local env is ok,I suspect something is wrong with the container
> > > > 
> > > > 
> > > > [This link may help you to fix client encoding error](https://h2c.tech/p/os%E5%AD%97%E7%AC%A6%E9%9B%86%E5%BD%B1%E5%93%8Dkingbase%E6%95%B0%E6%8D%AE%E5%BA%93%E8%BF%9E%E6%8E%A5/)
> > > 
> > > 
> > > I tried the method in the link, but I couldn't resolve the issue.It seems unrelated to the Kingbase database container.I used a local Spark for testing, but the Kingbase database was still the one in the container, and it worked successfully.
> > 
> > 
> > Yes, I have added the spark container, but it can't be solved.
> 
> <img alt="image" width="865" src="https://user-images.githubusercontent.com/72905543/253030175-2c577bdb-0c4e-4e97-af9c-9360d8885f06.png">

```shell
 docker run tyrantlucifer/spark:2.4.6 locale -a
 05:17:56.97 
 05:17:56.97 Welcome to the Bitnami spark container
 05:17:56.98 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-spark
 05:17:56.98 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-spark/issues

 05:17:56.98 
C
C.UTF-8
POSIX
```
This container doesn't support `zh_CN.utf-8`.
And the kingbase container(`huzhihui/kingbase:v8r6`) locale `en_US.UTF-8`.

CheneyYin commented:
![image](https://github.com/apache/seatunnel/assets/29588249/3a221d3e-3664-417e-b800-f45c66b65cd0)
It can fix `ansi_x3.4-1968` error.

NickCodeJourney commented:
> ![image](https://user-images.githubusercontent.com/29588249/253173019-3a221d3e-3664-417e-b800-f45c66b65cd0.png) It can fix `ansi_x3.4-1968` error.

thx sir

NickCodeJourney committed:
Create maven-publish.yml

Commit:
:fire:

NickCodeJourney commented:
<img width="512" alt="Snipaste_2023-07-13_22-25-58" src="https://github.com/apache/seatunnel/assets/72905543/bb4f0926-6f20-4dd8-bc68-f8a46cab6fc3">

Commit:
:memo: update java doc

hailin0 committed:
Merge branch 'dev' into dev

Commit:
:art: format code

NickCodeJourney committed:
Merge branch 'apache:dev' into dev

NickCodeJourney committed:
Merge branch 'apache:dev' into dev

Commit:
:art: format code

Commit:
:art: format code

NickCodeJourney committed:
Merge branch 'apache:dev' into dev

NickCodeJourney committed:
Merge branch 'apache:dev' into dev

NickCodeJourney committed:
Merge branch 'apache:dev' into dev

NickCodeJourney committed:
Merge branch 'apache:dev' into dev

liugddx commented:
@zhilinli123 PTAL

zhilinli123 commented:
Are you willing to add connector V2 documentation for this? 
Refer to code:https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/source/Mysql.md
Issues:https://github.com/apache/seatunnel/issues/4572
Issues example:https://github.com/apache/seatunnel/issues/4544
@NickYoungPeng

TyrantLucifer reviewed (commented):
Overall LGTM, PTAL @ic4y @liugddx

Commit:
:memo: add Kingbase doc

Commit:
Merge remote-tracking branch 'origin/dev' into dev

NickCodeJourney commented:
@zhilinli123 TBR

liugddx commented on docs/en/connector-v2/sink/Kingbase.md:
```diff
@@ -0,0 +1,159 @@
+
+## Description
+
+> Use `Xa transactions` to ensure `exactly-once`. So only support `exactly-once` for the database which is
+> support `Xa transactions`. You can set `is_exactly_once=true` to enable it.
```
@ic4y Does kingbase have this feature?

NickCodeJourney replied:
i will test the feature

NickCodeJourney replied:
> @ic4y Does kingbase have this feature?

sorry kingbase don't support the feature

NickCodeJourney replied:
`com.kingbase8.util.KSQLException: ERROR: prepared transactions are disabled
  Hint: Set max_prepared_transactions to a nonzero value.`
need Set max_prepared_transactions to a nonzero value

NickCodeJourney replied:
<img width="1313" alt="image" src="https://github.com/apache/seatunnel/assets/72905543/8a8367e9-e05b-42ac-9ad8-fa96b2f9e9f8">
i try it but not ok

NickCodeJourney replied:
don't support

liugddx commented on docs/en/connector-v2/sink/Kingbase.md:
```diff
@@ -0,0 +1,159 @@
+|------------|--------------------|----------------------|------------------------------------------|------------------------------------------------------------------------------------------------|
+| Kingbase   | v8r6               | com.kingbase8.Driver | jdbc:kingbase8://localhost:54321/db_test | [Download](https://repo1.maven.org/maven2/cn/com/kingbase/kingbase8/8.6.0/kingbase8-8.6.0.jar) |
+
+## Database Dependency
```
Add kingbase support versions? @zhilinli123

zhilinli123 replied:
> Add kingbase support versions? @zhilinli123

yes cc @liugddx @NickYoungPeng  https://github.com/apache/seatunnel/pull/5249

liugddx commented:
You need to put the screenshot of the e2e test here

zhilinli123 commented on docs/en/connector-v2/sink/Kingbase.md:
```diff
@@ -0,0 +1,159 @@
+# Kingbase
+
+> JDBC Kingbase Sink Connector
+
```
reference: https://github.com/apache/seatunnel/pull/5249

zhilinli123 commented on docs/en/connector-v2/sink/Kingbase.md:
```diff
@@ -0,0 +1,159 @@
+    row.num = 16
+    schema = {
+      fields {
+        name = "string"
```
Add a full type test

zhilinli123 replied:
such as: @NickYoungPeng 
```
source {
  FakeSource {
    row.num = 10
    map.size = 10
    array.size = 10
    bytes.length = 10
    string.length = 10
    schema = {
      fields {
        c_map = "map<string, array<int>>"
        c_array = "array<int>"
        c_string = string
        c_boolean = boolean
        c_tinyint = tinyint
        c_smallint = smallint
        c_int = int
        c_bigint = bigint
        c_float = float
        c_double = double
        c_decimal = "decimal(16, 1)"
        c_null = "null"
        c_bytes = bytes
        c_date = date
        c_timestamp = timestamp
      }
    }
    }
}
```

zhilinli123 commented on docs/en/connector-v2/source/Kingbase.md:
```diff
@@ -0,0 +1,145 @@
+# Kingbase
+
+> JDBC Kingbase Source Connector
+
```
Add version information as above

zhilinli123 commented on docs/en/connector-v2/source/Kingbase.md:
```diff
@@ -0,0 +1,145 @@
+    url = "jdbc:kingbase8://localhost:54321/db_test"
+    user = "root"
+    password = ""
+    compatible_mode = "mysql"
```
mykingbase?

Commit:
:bug: fix Kingbase doc

Commit:
:fire: delete Exactly-once

Commit:
:memo: Add a full type test

zhilinli123 commented:
Wait for ci

NickCodeJourney commented:
> You need to put the screenshot of the e2e test here

it's above and it's folded

NickYoungPeng commented:
> Wait for ci

last ci `Error:  Client bundle compiled with errors therefore further build is impossible.`

NickYoungPeng commented:
> Wait for ci

`SyntaxError: /home/runner/work/seatunnel/seatunnel/seatunnel-website/docs/connector-v2/sink/Kingbase.md: Expected corresponding JSX closing tag for <td>. (76:55)
Error:  Client bundle compiled with errors therefore further build is impossible.
  74 | </tr>
  75 | <tr parentName="tbody">
> 76 | <td parentName="tr" {...{"align":null}}>{`SMALLSERIAL`}</br>{` SERIAL `}</br>{`INT4`}</td>
     |                                                        ^
  77 | <td parentName="tr" {...{"align":null}}>{`INT`}</td>
  78 | </tr>
  79 | <tr parentName="tbody">
SyntaxError: /home/runner/work/seatunnel/seatunnel/seatunnel-website/docs/connector-v2/source/Kingbase.md: Expected corresponding JSX closing tag for <td>. (77:55)
  75 | </tr>
  76 | <tr parentName="tbody">
> 77 | <td parentName="tr" {...{"align":null}}>{`SMALLSERIAL`}</br>{` SERIAL `}</br>{`INT4`}</td>
     |                                                        ^
  78 | <td parentName="tr" {...{"align":null}}>{`INT`}</td>
  79 | </tr>
  80 | <tr parentName="tbody">
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! seatunnel-website@0.0.0 build: `docusaurus build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the seatunnel-website@0.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2023-08-10T02_06_00_437Z-debug.log
Error: Process completed with exit code 1.`

Commit:
:bug: fix doc

Commit:
:bug: fix doc

Commit:
:memo: Refine formatting in data type mapping tables

The formatting in the data type mapping tables for both the source and sink documents has been refined. This change mainly involves adjusting alignment and spacing to improve readability. No change in the content, only the format.

NickCodeJourney commented:
> You need to put the screenshot of the e2e test here

https://github.com/apache/seatunnel/pull/4803#issuecomment-1634345386
https://github.com/apache/seatunnel/pull/4803#issuecomment-1632798856

liugddx reviewed (approved):
LGTM

liugddx commented:
> LGTM

Please screenshot the test results. PTAL @ic4y

NickCodeJourney commented:
> 



> > LGTM
> 
> Please screenshot the test results. PTAL @ic4y

<img width="1727" alt="Snipaste_2023-08-15_11-19-09" src="https://github.com/apache/seatunnel/assets/72905543/83628a23-bebe-4031-9fb2-5e1c09c1e789">
<img width="1727" alt="Snipaste_2023-08-15_11-18-02" src="https://github.com/apache/seatunnel/assets/72905543/9ddf4c9d-452c-4ba3-ac87-8ff19babe849">

NickYoungPeng commented:
close #4758 #4512
